lib/swagger-mock-validator.ts

```diff
@@ -1,191 +1,259 @@
+import { SpecParser } from './swagger-mock-validator/spec-parser';
 import {
-    ValidationOutcome,
-    ValidationResult
-} from './api-types';
-import {Analytics} from './swagger-mock-validator/analytics';
-import {FileStore} from './swagger-mock-validator/file-store';
-import {MockParser} from './swagger-mock-validator/mock-parser';
-import {ParsedMock} from './swagger-mock-validator/mock-parser/parsed-mock';
-import {PactBroker} from './swagger-mock-validator/pact-broker';
-import {SpecParser} from './swagger-mock-validator/spec-parser';
-import {
-    MockSource,
-    ParsedSwaggerMockValidatorOptions,
-    SerializedMock,
-    SerializedSpec,
-    SpecSource,
-    SwaggerMockValidatorUserOptions,
-    ValidateOptions
+  MockSource,
+  ParsedSwaggerMockValidatorOptions,
+  SerializedMock,
+  SerializedSpec,
+  SpecSource,
+  SwaggerMockValidatorUserOptions,
+  ValidateOptions
 } from './swagger-mock-validator/types';
-import {validateSpecAndMock} from './swagger-mock-validator/validate-spec-and-mock';
+import { validateSpecAndMock } from './swagger-mock-validator/validate-spec-and-mock';
 
-const getMockSource = (mockPathOrUrl: string, providerName?: string): MockSource => {
-    if (providerName) {
-        return 'pactBroker';
-    } else if (FileStore.isUrl(mockPathOrUrl)) {
-        return 'url';
-    }
+const getMockSource = (
+  mockPathOrUrl: string,
+  providerName?: string
+): MockSource => {
+  if (providerName) {
+    return 'pactBroker';
+  } else if (FileStore.isUrl(mockPathOrUrl)) {
+    return 'url';
+  }
 
-    return 'path';
+  return 'path';
 };
 
-const getSpecSource = (specPathOrUrl: string): SpecSource => FileStore.isUrl(specPathOrUrl) ? 'url' : 'path';
+const getSpecSource = (specPathOrUrl: string): SpecSource =>
+  FileStore.isUrl(specPathOrUrl) ? 'url' : 'path';
 
-const parseUserOptions = (userOptions: SwaggerMockValidatorUserOptions): ParsedSwaggerMockValidatorOptions => ({
-    ...userOptions,
-    mockSource: getMockSource(userOptions.mockPathOrUrl, userOptions.providerName),
-    specSource: getSpecSource(userOptions.specPathOrUrl)
+// tslint:disable:cyclomatic-complexity
+const parseUserOptions = (
+  userOptions: SwaggerMockValidatorUserOptions
+): ParsedSwaggerMockValidatorOptions => ({
+  ...userOptions,
+  mockSource: getMockSource(
+    userOptions.mockPathOrUrl,
+    userOptions.providerName
+  ),
+  specSource: getSpecSource(userOptions.specPathOrUrl),
+  additionalPropertiesInResponse:
+    typeof userOptions.additionalPropertiesInResponse === 'undefined'
+      ? true // default to true - existing behaviour
+      : userOptions.additionalPropertiesInResponse === 'true'
+      ? true
+      : false,
+  requiredPropertiesInResponse:
+    typeof userOptions.requiredPropertiesInResponse === 'undefined'
+      ? false // default to false - existing behaviour
+      : userOptions.requiredPropertiesInResponse === 'true'
+      ? true
+      : false
 });
```

Comment:
This is where we take in the user captured input via commander, which comes in as `"undefined" | "false" | "true"` strings and either set our default behaviour (based on no value being provided, commander provides "undefined", or coerce the users selection into a bool.